src/topology/algebra/module/linear_pmap.lean

```diff
@@ -0,0 +1,179 @@
+import topology.algebra.module.basic
+
+/-!
+# Partially defined linear operators over topological vector spaces
+
+We define basic notions of partially defined linear operators.
+In this file we prove all elementary properties that do not assume that the underlying spaces
+are normed.
+
+## Main definitions
+
+* `linear_pmap.is_closed`: an operator is closed iff its graph is closed
+* `linear_pmap.is_closable`: an operator is closable iff the closure of its graph is a graph
+* `linear_pmap.closure`: the closure of a closable operator
+* `linear_pmap.has_core`: a submodule contained in the domain is a core if restricting to the core
+  does not lose information about the operator
+
+## Main statements
+
+* `linear_pmap.closable_iff_exists_closed_extension`: an operator is closable iff it has a closed
+  extension
+* `linear_pmap.closable.exists_unique`: there exists a unique closure
+* `linear_pmap.closure_has_core`: the domain of `f` is a core of its closure
+
+## References
+
+* [J. Weidmann, *Linear Operators in Hilbert Spaces*][weidmann_linear]
+
+## Tags
+
+Unbounded operators, closed operators
+-/
+
+
+open_locale topological_space
+
+variables {R E F 𝕜: Type*}
+
+variables [comm_ring R] [add_comm_group E] [add_comm_group F]
+variables [module R E] [module R F]
+variables [topological_space E] [topological_space F]
+
+namespace linear_pmap
+
+/-! ### Closed and closable operators -/
+
+/-- An operator is closed if its graph is closed. -/
+def is_closed (f : linear_pmap R E F) : Prop :=
+is_closed (f.graph : set (E × F))
+
+variables [has_continuous_add E] [has_continuous_add F]
+variables [topological_space R] [has_continuous_smul R E] [has_continuous_smul R F]
+
+/-- An operator is closable if the closure of the graph is a graph. -/
+def is_closable (f : linear_pmap R E F) : Prop :=
+∃ (f' : linear_pmap R E F), f.graph.topological_closure = f'.graph
+
+/-- A closed operator is trivially closable. -/
+lemma is_closed.is_closable {f : linear_pmap R E F} (hf : f.is_closed) : f.is_closable :=
+⟨f, hf.submodule_topological_closure_eq⟩
+
+/-- If `f` has a closable extension `g`, then `f` itself is closable. -/
+lemma is_closable.le_is_closable {f g : linear_pmap R E F} (hf : f.is_closable) (hfg : g ≤ f) :
+  g.is_closable :=
+begin
+  cases hf with f' hf,
+  have : g.graph.topological_closure ≤ f'.graph :=
+  by { rw ←hf, exact submodule.topological_closure_mono (le_graph_of_le hfg) },
+  refine ⟨g.graph.topological_closure.to_linear_pmap _, _⟩,
+  { intros x hx hx',
+    cases x,
+    exact f'.graph_fst_eq_zero_snd (this hx) hx' },
+  rw [submodule.to_linear_pmap_graph_eq],
+end
+
+/-- The closure is unique. -/
+lemma is_closable.exists_unique {f : linear_pmap R E F} (hf : f.is_closable) :
+  ∃! (f' : linear_pmap R E F), f.graph.topological_closure = f'.graph :=
+begin
+  refine exists_unique_of_exists_of_unique hf (λ _ _ hy₁ hy₂, eq_of_eq_graph _),
+  rw [←hy₁, ←hy₂],
+end
+
+open_locale classical
+
+/-- The closure of a closable operator. -/
```

Comment:
Could you specify in the docstring that when `f` is not closable we use `f` as a junk value?